RedisCore/Internal/Protocol/ProtocolHandler.cs

```diff
@@ -126,6 +126,8 @@ public static async ValueTask<RedisObject> Read(PipeReader reader, IBufferPool<b
         while (true)
         {
             var readResult = await reader.ReadAsync(cancellationToken);
+            // From this point we must read the content till the end otherwise we can't return socket to the pool 
+            cancellationToken = CancellationToken.None;
```

Comment:
Okay, I understand that it applies to the first `ReadAsync` operation as well from the second iteration of the while loop. Sounds like a bit dangerous (no way to cancel), but I guess this was a compromise.

Reply:
The point is that message must be read till the end if we were able to read at least one piece of it. This will let us to return the connection to the pool. Otherwise, if we cancel reading in the middle of the message, then connection can't be re-used